torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -253,6 +257,92 @@ def get_input_cpp_type(input):
             return DTYPE_TO_CPP[dtype]
         return f"ArrayRefTensor<{DTYPE_TO_CPP[input.get_dtype()]}>"
 
+    def generate_input_output_runtime_checks(self):
+        dtype_to_name = {
+            torch.bfloat16: "c10::BFloat16",
+            torch.float16: "c10::Half",
+            torch.float32: "float",
+            torch.float64: "double",
+            torch.bool: "bool",
+            torch.int8: "signed char",
+            torch.int16: "short int",
+            torch.int32: "int",
+            torch.int64: "long int",
+            torch.uint8: "unsigned char",
+            # the following dtype(s) are not supported by cpp_wrapper
+            # torch.uint16, torch.uint32, torch.uint64, torch.uint32
+            torch.complex32: "c10::complex<c10::Half>",
+            torch.complex64: "c10::complex<float>",
+            torch.complex128: "c10::complex<double>",
+            torch.float8_e4m3fn: "c10::Float8_e4m3fn",
+            torch.float8_e5m2: "c10::Float8_e5m2",
+            torch.float8_e4m3fnuz: "c10::Float8_e4m3fnuz",
+            torch.float8_e5m2fnuz: "c10::Float8_e5m2fnuz",
+        }
+
+        # In debug_compile mode, we generate checks to ensure the dtype/shape/stride of each
+        # real input/output tensor match ones provided at compile time via sample
+        # input/output.
+        def gen_check(handle_kind, idx, name, tensor):
+            self.prefix.writeline(f"auto {name} = {handle_kind}[{idx}];")
+            self.codegen_tensor_dtype_name_var_decl(self.prefix, name)
+            expected_dtype = dtype_to_name[tensor.dtype]
```

Comment:
I think we can check dtype by comparing enum values, not by strcmp.

Reply:
I tried to find an aten API to get the enum value for each dtype in C++ world, but couldn't. Particularly, we have aten tensor's ```dtype``` returns a ```TypeMeta``` object, which doesn't seem to provide an API to obtain the enum value for it:

```
https://caffe2.ai/doxygen-c/html/classat_1_1_tensor.html

https://caffe2.ai/doxygen-c/html/classcaffe2_1_1_type_meta.html
```

Reply:
I think `dtype().toScalarType()` should do it.

Reply:
Ah, Thanks! It turned out we actually have an existing shim function ```aoti_torch_get_dtype```. I ended up using this one.